openstack-mixin/panels.libsonnet

```diff
@@ -0,0 +1,1004 @@
+        targets=[t.securityGroups],
+        description='The number of network security groups managed by Neutron.',
+      )
+      + timeSeries.standardOptions.withDecimals(0),
+
+    neutronAgents:
+      table.new('Agents')
+      + table.queryOptions.withTargets([t.neutronAgentState])
+      + table.queryOptions.withDatasource('prometheus', '${prometheus_datasource}')
+      + table.panelOptions.withDescription('Details for the agents for OpenStack Neutron.')
+      + table.standardOptions.withOverridesMixin([
+        table.fieldOverride.byName.new('Admin state')
+        + table.fieldOverride.byName.withProperty('custom.displayMode', 'color-text')
+        + table.fieldOverride.byName.withPropertiesFromOptions(
+          table.standardOptions.withMappings(
+            {
+              type: 'value',
+              options: {
+                down: {
+                  color: 'red',
+                  index: 1,
+                  text: 'Down',
+                },
+                up: {
+                  color: 'green',
+                  index: 0,
+                  text: 'Up',
+                },
+              },
+            }
+          ),
+        ),
+      ])
+      + table.queryOptions.withTransformationsMixin([
+        {
+          id: 'joinByField',
+          options: {
+            byField: 'Time',
+            mode: 'outer',
+          },
+        },
+        {
+          id: 'organize',
+          options: {
+            excludeByName: {
+              Time: true,
+              Value: true,
+              __name__: true,
+              hostname: true,
+              job: true,
+            },
+            includeByName: {},
+            indexByName: {
+              Time: 6,
+              Value: 9,
+              __name__: 7,
+              adminState: 3,
+              hostname: 5,
+              id: 4,
+              instance: 1,
+              job: 8,
+              service: 0,
+              zone: 2,
+            },
+            renameByName: {
+              Time: '',
+              adminState: 'Admin state',
+              id: 'ID',
+              instance: 'Instance',
+              service: 'Service',
+              zone: 'Zone',
+            },
+          },
+        },
+      ]),
+
+    volumes:
+      commonlib.panels.generic.timeSeries.base.new(
+        'Volumes',
+        targets=[t.volumes],
+        description='The number of volumes managed by Cinder.',
+      )
+      + timeSeries.standardOptions.withDecimals(0),
+
+    volumeStatus:
+      commonlib.panels.generic.timeSeries.base.new(
+        'Volume status',
+        targets=[t.volumeErrorStatus, t.volumeNonErrorStatus],
+        description='The current status of volumes in Cinder.',
+      )
+      + timeSeries.options.legend.withPlacement('right')
+      + timeSeries.fieldConfig.defaults.custom.withStackingMixin({
+        group: 'A',
+        mode: 'normal',
+      })
+      + timeSeries.standardOptions.withDecimals(0),
+
+    volumeUsage:
```

Comment:
The legend looks very strange, can you recheck it?
<img width="829" alt="image" src="https://github.com/grafana/jsonnet-libs/assets/38274348/04f45d3c-4bf6-42dd-83d6-172b0224a713">

Reply:
See explanation below

Reply:
Any reason why the legend got moved to the bottom? Intentional?

Reply:
Not intentional, probably just forgot to add the option.

Reply:
Looks good to resolve